File: source/HttpInterface/WaveServerPageForRest.cpp
/***************************************************************************
 *   Copyright (C) 2005-2015 Vidyasagara Guntaka                           *
 *   All rights reserved.                                                  *
 *   Author : Vidyasagara Reddy Guntaka                                    *
 ***************************************************************************/

#include "HttpInterface/WaveServerPageForRest.h"
#include "HttpInterface/HttpInterfaceReceiverObjectManager.h"
#include "HttpToolKit.h"

namespace WaveNs
{

WaveServerPageForRest::WaveServerPageForRest (HttpInterfaceReceiverObjectManager *pHttpInterfaceReceiverObjectManager, const string &path)
    : WaveServerPage (pHttpInterfaceReceiverObjectManager, path)
{
}

WaveServerPageForRest::~WaveServerPageForRest ()
{
}

void WaveServerPageForRest::get (const HttpRequest &httpRequest)
{
    requestHandlerForGet (httpRequest);
}

void WaveServerPageForRest::requestHandlerForGet (const HttpRequest &httpRequest)
{
    string                 httpResponse;
    ServerStreamingSocket *pServerStreamingSocket = httpRequest.getPServerStreamingSocket ();

    HttpToolKit::getOkStringForGetForRestXml (httpResponse, "");

    if (NULL != pServerStreamingSocket)
    {
        (*pServerStreamingSocket) << httpResponse;
    }
}

}

File: Java/source/com/CxWave/Wave/HttpInterface/WaveServerPageForRest.java
/*************************************************************************************************************************
 * Copyright (C) 2015-2016 Vidyasagara Guntaka & CxWave, Inc * All rights reserved. * Author : Vidyasagara Reddy Guntaka *
 *************************************************************************************************************************/

package com.CxWave.Wave.HttpInterface;

import com.CxWave.Wave.Framework.Utils.Socket.AcceptedStreamingSocket;

public class WaveServerPageForRest extends WaveServerPage
{

    public WaveServerPageForRest (final HttpInterfaceReceiverObjectManager httpInterfaceReceiverObjectManager, final String path)
    {
        super (httpInterfaceReceiverObjectManager, path);
    }

    @Override
    public void get (final HttpRequest httpRequest)
    {
        requestHandlerForGet (httpRequest);
    }

    @Override
    public void requestHandlerForGet (final HttpRequest httpRequest)
    {
        final StringBuffer httpResponse = new StringBuffer ();
        final AcceptedStreamingSocket acceptedStreamingSocket = httpRequest.getAcceptedStreamingSocket ();

        HttpToolKit.getOkStringForGetForRestXml (httpResponse, "");

        if (null != acceptedStreamingSocket)
        {
            acceptedStreamingSocket.send (httpResponse);
        }
    }
}
